### PR TITLE
Update MonoMod to a daily build with Linux fixes

### PR DESCRIPTION
**Pull request by xIGBClutchIx**
Fixes issue #4090

### CODE REVIEW

**Comment by Chicken-Bones** [comment]
I'd love to have an official non daily version of this. Not being able to see the version on https://www.nuget.org/packages/MonoMod.RuntimeDetour#versions-body-tab or a commit on https://github.com/MonoMod/MonoMod/tree/reorganize is concerning

**Comment by xIGBClutchIx** [comment]
> I'd love to have an official non daily version of this. Not being able to see the version on https://www.nuget.org/packages/MonoMod.RuntimeDetour#versions-body-tab or a commit on https://github.com/MonoMod/MonoMod/tree/reorganize is concerning

Yeah, I was hoping to see a .1 version or something.

I saw the nuget config (https://github.com/tModLoader/tModLoader/blob/1.4.4/patches/tModLoader/Terraria/nuget.config) had the dev builds so thought I would use that as this is a decently big issue as we can't host a server on Linux currently.

I found the build in their artifacts (https://dev.azure.com/MonoMod/MonoMod/_artifacts/feed/DevBuilds/NuGet/MonoMod.RuntimeDetour/versions/)

**Comment by Chicken-Bones** [comment]
I have confirmed that `MonoMod.RuntimeDetour 25.1.0` contains the fix in https://github.com/MonoMod/MonoMod/issues/166
You can see the referenced commits for the build in https://nuget.info/packages/MonoMod.RuntimeDetour/25.1.0
I also confirmed that the code contained in https://github.com/MonoMod/MonoMod/commit/e045c407ec3619a3ade1b20d39eed72b1f07535e is present in the physical `MonoMod.Utils.dll` shipped in the current tModLoader stable release on steam.

In short, `MonoMod.RuntimeDetour 25.1.0` contains all the latest code, and if you are still experiencing this issue we will need to investigate it in #4090
